src/api/useFetch.js

```diff
@@ -19,6 +19,65 @@ import { FeatureRegistrationContext } from '../App'
 import { registerCallbackHandler } from '../feature'
 
 
+const useFetch = (
+  route,
+  params,
+  errorMsg,
+) => {
+  const [res, setRes] = React.useState()
+  const [isLoading, setIsLoading] = React.useState(true)
+  const [isError, setIsError] = React.useState()
+  const [error, setError] = React.useState()
+  const [locked, setLocked] = React.useState(false)
+
+  const { enqueueSnackbar } = useSnackbar()
+
+  React.useEffect(() => {
+    const fetch = async () => {
+      try {
+        setRes(await route(params))
+        setIsLoading(false)
+        setIsError(false)
+        setError(false)
+
+      } catch (e) {
+        setIsLoading(false)
+        setIsError(true)
+        setError(e.toString())
+
+        enqueueSnackbar(
```

Comment:
make snackbar optional (we might not _always_ want it)?

Reply:
I think we always want it?
Not only is it transparent to user that _something_ went wrong, it also offers a retry

still, we can make it optional even though it is always used (for now)